### PR TITLE
gather_facts network platforms

### PR DESCRIPTION
**Pull request by trishnaguha**
##### SUMMARY

gather_facts network platforms
This PR will only get in if the PR https://github.com/ansible/ansible/pull/49399 does **not** get in 2.8

**Platforms:**
EOS
IOS
IOSXR
NXOS
VYOS
JUNOS
FRR https://github.com/ansible/ansible/pull/53444 Add FRR collector [WIP]

##### ISSUE TYPE

- Feature Pull Request

### CODE REVIEW

**Comment by abenokraitis** [comment]
CC @abenokraitis @IPvSean

**Comment by mattclay** [comment]
CI failure in unit tests: https://app.shippable.com/github/ansible/ansible/runs/102374/9/tests

**Comment by trishnaguha** [comment]
@mattclay CI is passing now.

File: lib/ansible/module_utils/facts/collector.py [hunk 1]
```diff
@@ -356,7 +359,11 @@ def collector_classes_from_gather_subset(all_collector_classes=None,
 
     minimal_gather_subset = minimal_gather_subset or frozenset()
 
-    platform_info = platform_info or {'system': platform.system()}
+    if connection:
+        capabilities = json.loads(connection.get_capabilities())
```
Comment (jctanner) [inline lib/ansible/module_utils/facts/collector.py:363]: Seems like we need some validation of the object type for "connection". It's also not clear where "capabilities" are set from this patch alone. Is that config / variables? Is there some sort of input validation happening elsewhere that warns when something is missing or wrong?

Comment (trishnaguha) [reply lib/ansible/module_utils/facts/collector.py:363]: fixed by https://github.com/ansible/ansible/commit/9118c5ff7633f732affb0bd5b41dcd0359f677d7
capabilities is a JSON data format that contains network device information. It always exists if the connection object has get_capabilities method.

**Comment by trishnaguha** [comment]
cc @cidrblock

**Comment by trishnaguha** [comment]
Close this PR as https://github.com/ansible/ansible/pull/49399 is merged.

**Pull request closed without merging**